docs/source/faq.rst

```diff
@@ -0,0 +1,166 @@
+
+How can I debug a failing test?
+-------------------------------
+
+Short of using a bash debugger you should make sure to use appropriate asserts for your task instead of raw bash comparisons, e.g.:
+
+.. code-block:: bash
+
+    @test test {
+        run echo test failed
+        assert_output "test"
+        # instead of 
+        [ "$output" = "test" ]
+    }
+
+Because the former will print the output when the test fails while the latter won't.
+Similarly, you should use `assert_success`/`assert_failure` instead of `[ "$status" -eq 0]` for return code checks.
+
+Is there a mechanism to add file/test specific functionality to a common setup function?
+----------------------------------------------------------------------------------------
+
+Often the setup consists of parts that are common between different files of a test suite and parts that are specific to each file.
+There is no suite wide setup functionality yet, so you should extract these common setup steps into their own file (e.g. `common-test-setup.sh`) and function (e.g. `commonSetup() {}`),
+which can be `source`d or `load`ed and call it in `setup_file` or `setup`.
+
+How can I use helper libraries like bats-assert?
+------------------------------------------------
+
+This is a short reproduction of https://github.com/ztombol/bats-docs.
+
+At first, you should make sure the library is installed. This is usually done in the `test_helper/` folders alongside the `.bats` files, giving you a filesystem layout like this:
+
+.. code-block::
+
+    test/
+        test.bats
+        test_helper/
+            bats-support/
+            bats-assert/
+
+Next, you should load those helper libraries:
+
+.. code-block:: bash
+
+    setup() {
+        load 'test_helper/bats-support/load' # this is required by bats-assert!
+        load 'test_helper/bats-assert/load'
+    }    
+
+Now, you should be able to use the functions from these helpers inside your tests, e.g.:
+
+.. code-block:: bash
+
+    @test "test" {
+        run echo test
+        assert_output "test"
+    }
+
+Note that you obviously need to load the library before using it.
+If you need the library inside `setup_file` or `teardown_file` you need to load it in `setup_file`.
+
+How to set a test timeout in bats?
+----------------------------------
+
+Unfortunately, this is not possible yet. Please contribute to issue `#396 <https://github.com/bats-core/bats-core/issues/396>`_ for further progress.
+
+How can I lint/shell-format my bats tests?
+------------------------------------------
+
+Due to their custom syntax (`@test`), `.bats` files are not standard bash. This prevents most tools from working with bats.
+However, shellcheck supports `.bats` files since version 0.7 and there is an alternative syntax `function_name { # @test` to declare tests in a bash compliant manner.
+There is currently no known formatter that supports formatting custom `.bats` syntax files directly.
```

Comment:
The [shfmt](https://github.com/mvdan/sh/blob/b8bd8e57547725220dbf237072b80e40c09a0502/cmd/shfmt/shfmt.1.scd#L49)  should be able to do this.

> Language variant to parse (bash/posix/mksh/bats, default "bash").

Reply:
Nice find, thank you for the review.